src/Roundcube/Composer/ExtensionInstaller.php

```diff
@@ -29,12 +29,12 @@ class ExtensionInstaller extends LibraryInstaller
      */
     public function getInstallPath(PackageInterface $package)
     {
-        static $vendorDir;
-        if ($vendorDir === null) {
-            $vendorDir = $this->getVendorDir();
+        static $vendorDir = [];
```

Comment:
Currently, `$this->getVendorDir()` returns `getcwd()`.

I belive we should use composer API instead and not cache the result at all (or cache it using all the variables that were used to parametrize the API).

Reply:
> Currently, `$this->getVendorDir()` returns `getcwd()`.

In this class. But it's being overridden by both the `PluginInstaller` and the `SkinInstaller` subclasses, which do the actual work of plugin installation.

Reply:
Again, there is `getcwd()` which is fragile. It would be much better to query composer API where `roundcube/roundcube` package is. Such result will be guaranteed to be "script runtime stable" and IMO fast enough to not need any caching.

Reply:
Based on https://github.com/phpstan/extension-installer/blob/1.3.1/src/Plugin.php#L137 it should be easy as:
```
$composer = $event->getComposer();
$installationManager = $composer->getInstallationManager();

return $installationManager->getInstallPath('roundcube/roundcube');
```

Reply:
It's `roundcube/roundcubemail` and it does not exist if you didn't install Roundcube as a dependency. SO, that wouldn't work. Am I right?

Reply:
Thanks to https://github.com/roundcube/plugin-installer/pull/44 this is always satisfied. Root-project (in sense of composer terminology, ie. installed as main project/not as dependency) can be analysed using composer API perfectly as well :).

Reply:
@mvorisek Thanks for the tip with the composer API, I will take a look.

Reply:
Will be fixed in #46 - see https://github.com/roundcube/plugin-installer/commit/81c25a273e8d2b360589924db74191470dea2b06 and https://github.com/roundcube/plugin-installer/commit/d79bd11297ac7ecc6ca383e54d91410898913134. Can you please test/review?